akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.{ Http, server }
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+
+import scala.io.StdIn
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  def main(args: Array[String]) = {
+    implicit val system = actorSystem()
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+    postStartActorSystem(system)
+
+    val bindingFuture = Http().bindAndHandle(
+      handler = route,
+      interface = config.host,
+      port = config.port,
+      settings = config.serverSettings.getOrElse(ServerSettings(system)))
+
+    println(s"Server online at http://${config.host}:${config.port}/\nPress RETURN to stop...")
+    StdIn.readLine()
+
+    bindingFuture
+      .flatMap(_.unbind())
+      .onComplete(_ ⇒ {
+        preTerminateActorSystem(system)
+        system.terminate()
+      })
+  }
+
+  /**
+   * Provides the [[ActorSystem]] from where the routes will be served. Override this method, if the default [[ActorSystem]] doesn't suit your needs. You don't need to store the created instance of the [[ActorSystem]].
+   */
+  def actorSystem(): ActorSystem = ActorSystem("my-system")
+
+  /**
+   * Hook that will be called once the actor system is created. Override this method if you want to do something with this [[ActorSystem]] like creating actors.
+   */
+  def postStartActorSystem(actorSystem: ActorSystem): Unit = {
+  }
+
+  /**
+   * Hook that will be called just before the termination of the actor system. Override this methods if you want to perform some actions on the current actorSystem, like gracefully shutting down any created actor.
+   */
+  def preTerminateActorSystem(actorSystem: ActorSystem): Unit = {
+  }
+
+  /**
+   * Your class should provide an implementation for the routes. These will be bound to the [[Http]]
+   */
+  val route: server.Route
+
+  /**
+   * Provides configuration values for the host, port and [[ServerSettings]] on which the server will run
+   */
+  val config = Config("localhost", 8080, None)
```

Comment:
If the goal is to remove boilerplate, would it make sense to read these from a Typesafe Config or env variable so they can be overridden from the command line, e.g. `-Dakka.http.server.app.port=3000`?